scripts/mbedtls_framework/c_parsing_helper.py

```diff
@@ -129,3 +153,17 @@ def read_function_declarations(functions: Dict[str, FunctionInfo],
                                        return_type,
                                        name,
                                        arguments)
+
+_C_TYPEDEF_DECLARATION_RE = r'typedef (?:struct ){0,1}(?P<type>\w+) (?P<name>\w+)'
+
+def read_typedefs(filename: str)->  Dict[str, str]:
```

Comment:
This only captures some typedefs. Why those? What will happen if we edit a header and the typedef no longer follows that format, and how much of a risk is there that this happens?

I don't know what this function is meant to be used for, and currently it isn't used in #25 or Mbed-TLS/mbedtls#9224.

Reply:
Reading simple typedefs is something that could be usefull for a `FunctionInfo` parsing class. 

It is currently not being used, but will be used when parsing psa headers for generating serialisation logic. It can also be very usefull for debugging purposes since it generates a simple to read dictionary e.g

    {'mbedtls_svc_key_id_t': 'psa_key_id_t',
     'psa_algorithm_t': 'uint32_t',
     'psa_dh_family_t': 'uint8_t',
     'psa_ecc_family_t': 'uint8_t',
     'psa_key_attributes_t': 'psa_key_attributes_s',
     'psa_key_derivation_step_t': 'uint16_t',
     'psa_key_id_t': 'uint32_t',
     'psa_key_lifetime_t': 'uint32_t',
     'psa_key_location_t': 'uint32_t',
     'psa_key_persistence_t': 'uint8_t',
     'psa_key_production_parameters_t': 'psa_key_production_parameters_s',
     'psa_key_slot_number_t': 'uint64_t',
     'psa_key_type_t': 'uint16_t',
     'psa_key_usage_t': 'uint32_t',
     'psa_status_t': 'int32_t'}

There is very little risk, since we don't change psa headers that much, and evem when we do, the data structures are very rarely updated.  Also this will be used in test code generation so whoever uses it needs to make sure it is capturing everything and update it accordingly. 

For now the simple parsing should work.